src/core/memory.h

```diff
@@ -2,3 +2,10 @@ int      lock_memory();
 void    *allocate_huge_page(int size);
 uint64_t phys_page(uint64_t virt_page);
 
+void setup_signal();
+
+
+struct map_ids_t {
+   int huge_page_bits;
+   int ids[16384];
```

Comment:
I'd like to know what does it depend on... for a while I dabbled on giving it an extra offset, since it seems to pick a semi-random point and then grow downwards.  But then I realized that any unused page-size segment of this array wouldn't get paged in, so by giving it an outrageous size we only waste hard disk space, not RAM.  So, it wouldn't be crazy to just set it at 8*1024*1024 to cover the whole segment.